public/app/features/dashboard/state/analyticsProcessor.ts

```diff
@@ -1,10 +1,16 @@
 import { getDashboardSrv } from '../services/DashboardSrv';
+import { DashboardModel } from './DashboardModel';
 
 import { PanelData, LoadingState, DataSourceApi } from '@grafana/data';
 
-import { reportMetaAnalytics, MetaAnalyticsEventPayload } from '@grafana/runtime';
+import {
+  reportMetaAnalytics,
+  MetaAnalyticsEventName,
+  DataRequestEventPayload,
+  DashboardViewEventPayload,
+} from '@grafana/runtime';
 
-export function getAnalyticsProcessor(datasource: DataSourceApi) {
+export function handleDataRequestEvent(datasource: DataSourceApi) {
```

Comment:
the `handle` prefix feels wrong to me here. This is basically an emit function that broadcasts an event (so almost the inverse of handle function). 

so rename handle to emit would make more sense here

Reply:
Right, I renamed them, thanks!